_posts/2020-12-17-solidity-v0.8.0-release-announcement.md

```diff
@@ -0,0 +1,237 @@
+7. 0x41: If you allocate too much memory or create an array that is too large.
+8. 0x51: If you call a zero-initialized variable of internal function type.
+
+This list of codes can be extended in the future.
+
+Note that you currently cannot use ``try ... catch Panic(uint _code) { ... }`` to catch a panic,
+but this is planned for the near future.
+
+#### Actual Checked Arithmetic
+
+By default, all arithmetic operations will perform overflow and underflow checks
+(Solidity already had division by zero checks). In case of an underflow or
+overflow, a ``Panic(0x11)`` error will be thrown and the call will revert.
+
+The following code, for example, will trigger such an error:
+
+```solidity
+contract C {
+    function f() public pure {
+        uint x = 0;
+        x--;
+    }
+}
+```
+
+The checks are based on the actual type of the variable, so even if the result would fit
+an EVM word of 256 bits, if your type is ``uint8`` and if the result is larger than ``255``,
+it will trigger a ``Panic``.
+
+We did not introduce checks for explicit type conversions from larger to smaller
+types, because we think that such checks might be unexpected. We would love to get
+your input on that matter!
+
+The following operations now have checks that they did not have before:
+ - addition (``+``), subtraction (``-``), multiplication (``*``)
+ - increment and decrement (``++`` / ``--``)
+ - unary negation (``-``)
+ - exponentiation (``**``)
+ - division (see below) (``/``)
+
+There are some edge cases you might not expect.
+For example, unary negation is only available for signed types.
+The problem is that in two's complement representation, there is one more negative number for each bit width
+than there are positive numbers. This means that the following code will trigger a ``Panic``:
+
+```solidity
+function f() pure {
+    int8 x = -128;
+    -x;
+}
+```
+
+The reason is that ``128`` does not fit an 8-bit signed integer.
+
+For the same reason, signed division can result in an assertion:
+```solidity
+function f() pure {
+    int8 x = -128;
+    x/(-1);
+}
+```
+
+Everyone who wanted to write a ``SafeMath`` function for exponentiation probably noticed
+that it is rather expensive to do that because the EVM does not provide overflow signalling.
+Essentially you have to implement your own ``exp`` routine without using the ``exp`` opcode
+for the general case.
+
+We hope that we found a rather efficient implementation and would also appreciate your feedback about that!
+
+For many special cases, we actually implemented it using the ``exp`` opcode instead of our own implementation.
+More specifically, exponentiation operations that use a literal number as base will use the ``exp`` opcode directly.
+There are also specialized code paths for bases that are variables with small value:
+All bases up to 2, bases between 3 and 10 and bases between 11 and 306 get special treatment
```

Comment:
```suggestion
all bases up to 2, bases between 3 and 10 and bases between 11 and 306 get special treatment
```

Reply:
It's a full sentence, so I think capital `A` is correct.

Reply:
TIL! https://www.grammarly.com/blog/capitalization-after-colons/

I always assumed that a colon does not end the sentence but apparently that's another part of the language where opinions vary.